sqlmesh/core/table_diff.py

```diff
@@ -322,6 +337,47 @@ def name(e: exp.Expression) -> str:
                 stats_df["t_only_count"] = stats_df["t_count"] - stats_df["join_count"]
                 stats = stats_df.iloc[0].to_dict()
 
+                if check_grain:
```

Comment:
can this be included in the main query without having to do another query?

Reply:
can't we just do a count distinct?

Reply:
Sure, I wanted to keep it in a separate query, to control it by the flag, but I will try to add it in the main query

Reply:
> Sure, I wanted to keep it in a separate query, to control it by the flag

You can still control it using the flag and append to the existing query instead of making a new one, e.g. either by mutating it or just calling `select` on it and just overwriting the local variable.

Reply:
Yes this is how I currently have it so that is in the main query, I will push this implementation, when I address the rest of your suggestions as well